api/genericresource/helpers.go

```diff
@@ -0,0 +1,35 @@
+package genericresource
```

Comment:
The NewError isn't used anywhere, the others are as the mask the type generated by the oneof